src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -572,13 +575,23 @@ static QVariant fcnAggregate( const QVariantList &values, const QgsExpressionCon
     if ( context && context->hasCachedValue( cacheKey ) )
       return context->cachedValue( cacheKey );
 
-    QgsExpressionContext subContext( *context );
-    QgsExpressionContextScope *subScope = new QgsExpressionContextScope();
-    subScope->setVariable( QStringLiteral( "parent" ), context->feature() );
-    subContext.appendScope( subScope );
-    result = vl->aggregate( aggregate, subExpression, parameters, &subContext, &ok );
+    if ( context->indexOfScope( "Symbol scope" ) != -1 )
+    {
+      //QgsFeatureIds *fids = context->variable( "symbol_feature_ids" )
+      QgsExpressionContext context_copy = *context;
+      const QgsFeatureIds fids = vl->countSymbolFeatures()->featureIds( context_copy.variable( "symbol_id" ).toString() );
+      result = vl->aggregate( aggregate, subExpression, parameters, &context_copy, &ok, fids );
+    }
+    else
+    {
+      QgsExpressionContext subContext( *context );
+      QgsExpressionContextScope *subScope = new QgsExpressionContextScope();
+      subScope->setVariable( QStringLiteral( "parent" ), context->feature() );
+      subContext.appendScope( subScope );
+      result = vl->aggregate( aggregate, subExpression, parameters, &subContext, &ok );
```

Comment:
This part of the code check if the symbol scope is in the current scopes, if it is, then the list of feature ids will be provided to the aggregate function in order to operate only on the subset of features associated with that symbol.